application/modules/front/controllers/CollectionController.php

```diff
@@ -236,6 +236,10 @@ public function actionDispbook($collectionName, $ourBookID, $hadithNumbers = NUL
 			
 			if (isset($previousURN) && !is_null($previousURN))
 				$viewVars['previousPermalink'] = $this->util->getPermalinkByURN($previousURN, "english");
+
+			// Add canonical links to single pages only
+			if ( $permalinkCanonical && count($pairs) === 1)
```

Comment:
Can we find another way to ensure it's a single hadith view page instead of `count($pairs)`? Sometimes books have one hadith in them.